vraptor-core/src/test/java/br/com/caelum/vraptor/view/DefaultRefererResultTest.java

```diff
@@ -150,4 +150,89 @@ public void whenRefererMatchesAControllerShouldForwardToIt() throws Exception {
+		when(request.getContextPath()).thenReturn("/vrap");
+		when(router.parse("/anything/ok", HttpMethod.GET, request)).thenReturn(method);
+		
+		doReturn(logic).when(result).use(logic());
+		when(logic.redirectTo(RefererController.class)).thenReturn(controller);
+		
+		refererResult.redirect();
+		
+		verify(logic).redirectTo(RefererController.class);
+		verify(controller).index();
+	}
+
+	@Test
+	public void whenRefererContainsCtxPathStrInTheHostItShouldForwardCorrectly() throws Exception {
+		LogicResult logic = mock(LogicResult.class);
+		RefererController controller = mock(RefererController.class);
+		
+		Method index = RefererController.class.getMethod("index");
+		ControllerMethod method = DefaultControllerMethod.instanceFor(RefererController.class, index);
+		
+		when(request.getHeader("Referer")).thenReturn("http://vraptor.test.com/vrap/anything/ok");
+		when(request.getContextPath()).thenReturn("/vrap");
+		when(router.parse("/anything/ok", HttpMethod.GET, request)).thenReturn(method);
+		
+		doReturn(logic).when(result).use(logic());
+		when(logic.forwardTo(RefererController.class)).thenReturn(controller);
+		
+		refererResult.forward();
+		
+		verify(logic).forwardTo(RefererController.class);
+		verify(controller).index();
+	}
+	
+	@Test
+	public void whenRefererDoesNotContainsCtxPathItShouldRedirectCorrectly() throws Exception {
+		LogicResult logic = mock(LogicResult.class);
+		RefererController controller = mock(RefererController.class);
+		
+		Method index = RefererController.class.getMethod("index");
+		ControllerMethod method = DefaultControllerMethod.instanceFor(RefererController.class, index);
+		
+		when(request.getHeader("Referer")).thenReturn("http://vraptor.test.com/anything/ok");
+		when(request.getContextPath()).thenReturn("/vrap");
+		when(router.parse("/anything/ok", HttpMethod.GET, request)).thenReturn(method);
+		
+		doReturn(logic).when(result).use(logic());
+		when(logic.redirectTo(RefererController.class)).thenReturn(controller);
+		
+		refererResult.redirect();
+		
+		verify(logic).redirectTo(RefererController.class);
+		verify(controller).index();
+	}
+	
+	@Test
+	public void whenRefererDoesNotContainsCtxPathItShouldForwardCorrectly() throws Exception {
+		LogicResult logic = mock(LogicResult.class);
+		RefererController controller = mock(RefererController.class);
+		
+		Method index = RefererController.class.getMethod("index");
+		ControllerMethod method = DefaultControllerMethod.instanceFor(RefererController.class, index);
+		
+		when(request.getHeader("Referer")).thenReturn("http://vraptor.test.com/anything/ok");
+		when(request.getContextPath()).thenReturn("/vrap");
+		when(router.parse("/anything/ok", HttpMethod.GET, request)).thenReturn(method);
+		
+		doReturn(logic).when(result).use(logic());
+		when(logic.forwardTo(RefererController.class)).thenReturn(controller);
+		
+		refererResult.forward();
+		
+		verify(logic).forwardTo(RefererController.class);
+		verify(controller).index();
+	}
```

Comment:
Well, that was my first commit on VRaptor so I tried to write comprehensive unit tests :-) - but I agree it's too much for something so simple. In addition it's a hard condition to achieve, the strings must match in a little-prob coincidence.

About the extension point, I don't have an opinion - would you expect this class to be extended? If so, then yes it's a good way to maintain its structure (or else people would have to create another method to perform this check, or even put its code within the callers)